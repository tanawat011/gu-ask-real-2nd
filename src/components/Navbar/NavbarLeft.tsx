import tw, { styled } from 'twin.macro'

import { IconFa } from 'components/Icons'
import { BORDER_THEME } from 'constants/twTheme'

export type NavbarLeftProps = {
  handleToggleSidebar: () => void
}

const TwNavbarLeft = tw.div`flex w-full`
const TwNavbarSearchInput = styled.input(() => [
  BORDER_THEME.BASE,
  tw`mx-1 w-full rounded-lg border bg-gray-800 px-3 py-2 text-sm text-white outline-none focus-within:(border-majorelle-blue ring-majorelle-blue) focus:(border-majorelle-blue ring-1 ring-majorelle-blue)`,
])

const TwWrapBurger = tw.div`mx-1 cursor-pointer py-2 px-3 hover:(rounded-full bg-black bg-opacity-40)`

export const NavbarLeft: React.FC<NavbarLeftProps> = ({
  handleToggleSidebar,
}) => {
  return (
    <TwNavbarLeft>
      <TwWrapBurger onClick={handleToggleSidebar}>
        <IconFa icon='bars' size='lg' />
      </TwWrapBurger>

      <TwNavbarSearchInput placeholder='Search...' />
    </TwNavbarLeft>
  )
}
