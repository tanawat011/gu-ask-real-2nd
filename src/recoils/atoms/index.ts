export * from './pageUiComponents'
export * from './localSetting'
export * from './sidebar'
