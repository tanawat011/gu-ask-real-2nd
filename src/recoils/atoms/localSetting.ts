import type { Language } from 'locales'
import type { TwColor, TwColorLevel } from 'types'

import { atom } from 'recoil'

export type LocalSettingState = {
  theme: {
    color: TwColor
    direction: 'ltr' | 'rtl'
    mode: 'light' | 'dark'
    locale: Language
    colorLevel: TwColorLevel
    panelExpand: boolean // config menu
    navMode: 'transparent' | 'themed'
    layout: {
      type: 'classic' | 'modern' | 'stack-side' | 'simple' | 'decked' | 'blank'
      sideNavCollapse: boolean
    }
  }
  auth: {
    session: {
      token: string
      signedIn: boolean
    }
    user: {
      avatar: string
      name: string
      email: string
      authority: string[] // ['admin', 'user']
    }
  }
}

export const defaultLocalSetting: LocalSettingState = {
  theme: {
    color: 'indigo',
    direction: 'ltr',
    mode: 'dark',
    locale: 'en',
    colorLevel: '600',
    panelExpand: false,
    navMode: 'transparent',
    layout: {
      type: 'modern',
      sideNavCollapse: true,
    },
  },
  auth: {
    session: {
      token: '',
      signedIn: true,
    },
    user: {
      avatar: '/img/avatars/thumb-1.jpg',
      name: 'Carolyn Perkins',
      email: '',
      authority: ['admin', 'user'],
    },
  },
}

export const localSettingAtom = atom<LocalSettingState>({
  key: 'localSettingAtom',
  default: defaultLocalSetting,
})
