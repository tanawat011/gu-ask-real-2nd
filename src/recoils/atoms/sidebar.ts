import { atom } from 'recoil'

export type SidebarState = {
  fullSidebar: boolean
}

export const sidebarAtom = atom<SidebarState>({
  key: 'sidebarAtom',
  default: {
    fullSidebar: true,
  },
})
