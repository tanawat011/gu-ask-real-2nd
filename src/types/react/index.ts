export * from './children'
export * from './reactRouterDom'
