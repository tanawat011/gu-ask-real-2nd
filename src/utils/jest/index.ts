export * from './renderWithProviders'
export * from './twColor'
