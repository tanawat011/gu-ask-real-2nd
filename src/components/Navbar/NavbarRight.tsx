import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { US } from 'country-flag-icons/react/3x2'
import tw from 'twin.macro'

const TwIconProfile = tw.div`h-8 w-8 rounded-full bg-cadet-grey`

const TwNavbarRight = tw.div`flex w-full items-center justify-end`
const TwNavbarItem = tw.div`mx-1 cursor-pointer p-2 hover:(rounded-full bg-black bg-opacity-40)`

const TwWrapIcon = tw.div`flex h-6 w-6 items-center justify-center`

const TwNavbarProfileContainer = tw.div`ml-3 flex cursor-pointer items-center pr-3`
const TwNavbarProfileWrap = tw.div`ml-2 flex select-none flex-col`
const TwNavbarProfileRole = tw.span`text-xs`
const TwNavbarProfileName = tw.span`font-bold text-sm`

export const NavbarRight = () => {
  return (
    <TwNavbarRight>
      <TwNavbarItem>
        <US className='h-6 w-6' />
      </TwNavbarItem>

      <TwNavbarItem>
        <TwWrapIcon>
          <FontAwesomeIcon icon='bell' size='lg' />
        </TwWrapIcon>
      </TwNavbarItem>

      <TwNavbarItem>
        <TwWrapIcon>
          <FontAwesomeIcon icon='gear' size='lg' />
        </TwWrapIcon>
      </TwNavbarItem>

      <TwNavbarProfileContainer>
        <TwIconProfile />
        <TwNavbarProfileWrap>
          <TwNavbarProfileRole>Admin</TwNavbarProfileRole>

          <TwNavbarProfileName>Tanawat P</TwNavbarProfileName>
        </TwNavbarProfileWrap>
      </TwNavbarProfileContainer>
    </TwNavbarRight>
  )
}
