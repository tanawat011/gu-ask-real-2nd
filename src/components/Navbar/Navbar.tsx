import type { SidebarState } from 'recoils/atoms'

import { useRecoilValue } from 'recoil'
import tw, { styled } from 'twin.macro'

import { sidebarAtom } from 'recoils/atoms'

import { NavbarLeft } from './NavbarLeft'
import { NavbarRight } from './NavbarRight'

export type NavbarProps = {
  handleToggleSidebar: () => void
}

const TwNavbar = styled.div(({ fullSidebar }: SidebarState) => [
  tw`flex h-sidebar-header items-center justify-between border-charcoal border-b px-4`,
  !fullSidebar && tw`sticky top-0 z-10 bg-yankees-blue`,
])

export const Navbar: React.FC<NavbarProps> = ({ handleToggleSidebar }) => {
  const { fullSidebar } = useRecoilValue(sidebarAtom)

  return (
    <TwNavbar fullSidebar={fullSidebar}>
      <NavbarLeft handleToggleSidebar={handleToggleSidebar} />

      <NavbarRight />
    </TwNavbar>
  )
}
