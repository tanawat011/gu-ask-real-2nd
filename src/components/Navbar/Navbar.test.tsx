import { useEffect } from 'react'

import { useRecoilState } from 'recoil'

import { sidebarAtom } from 'recoils/atoms'
import { renderWithProviders } from 'utils/jest'

import { Navbar } from './Navbar'

jest.mock('./Navbar', () => jest.requireActual('./Navbar'))

const RecoilComponent = ({
  fullSidebar = false,
}: {
  fullSidebar?: boolean
}) => {
  const [, setSidebar] = useRecoilState(sidebarAtom)

  useEffect(() => {
    setSidebar({ fullSidebar })
  }, [])

  return null
}

describe('<Navbar />', () => {
  test('should render without crashing and match snapshot', () => {
    expect.hasAssertions()

    const { asFragment } = renderWithProviders(
      <>
        <RecoilComponent fullSidebar />
        <Navbar handleToggleSidebar={jest.fn()} />
      </>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render without crashing with mini sidebar', () => {
    expect.hasAssertions()

    const { container } = renderWithProviders(
      <>
        <RecoilComponent />
        <Navbar handleToggleSidebar={jest.fn()} />
      </>,
    )

    expect(container).toBeInTheDocument()
  })
})
