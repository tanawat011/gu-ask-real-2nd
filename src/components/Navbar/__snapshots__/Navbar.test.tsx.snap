// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render without crashing and match snapshot 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c3 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
  width: 100%;
  border-radius: 0.5rem;
  border-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgb(55 65 81 / var(--tw-border-opacity));
  --tw-bg-opacity: 1;
  background-color: rgb(31 41 55 / var(--tw-bg-opacity));
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
  outline: 2px solid transparent;
  outline-offset: 2px;
}

.c3:focus-within {
  --tw-border-opacity: 1;
  border-color: rgb(79 70 229 / var(--tw-border-opacity));
  --tw-ring-opacity: 1;
  --tw-ring-color: rgb(79 70 229 / var(--tw-ring-opacity));
}

.c3:focus {
  --tw-border-opacity: 1;
  border-color: rgb(79 70 229 / var(--tw-border-opacity));
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(1px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow),var(--tw-ring-shadow),var(--tw-shadow,0 0 #0000);
  --tw-ring-opacity: 1;
  --tw-ring-color: rgb(79 70 229 / var(--tw-ring-opacity));
}

.c2 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
  cursor: pointer;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
}

.c2:hover {
  border-radius: 9999px;
  --tw-bg-opacity: 0.4;
  background-color: rgb(0 0 0 / var(--tw-bg-opacity));
}

.c9 {
  height: 2rem;
  width: 2rem;
  border-radius: 9999px;
  --tw-bg-opacity: 1;
  background-color: rgb(156 163 175 / var(--tw-bg-opacity));
}

.dark .c8 {
  --tw-bg-opacity: 1;
  background-color: rgb(239 68 68 / var(--tw-bg-opacity));
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c5 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
  cursor: pointer;
  padding: 0.5rem;
}

.c5:hover {
  border-radius: 9999px;
  --tw-bg-opacity: 0.4;
  background-color: rgb(0 0 0 / var(--tw-bg-opacity));
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  width: 1.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  margin-left: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 0.75rem;
}

.c10 {
  margin-left: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  font-size: 0.75rem;
  line-height: 1rem;
}

.c12 {
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 700;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 64px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom-width: 1px;
  --tw-border-opacity: 1;
  border-color: rgb(55 65 81 / var(--tw-border-opacity));
  padding-left: 1rem;
  padding-right: 1rem;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <svg>
          bars
        </svg>
      </div>
      <input
        class="c3"
        placeholder="Search..."
      />
    </div>
    <div
      class="c4"
    >
      <div
        class="c5 select-none"
      >
        Toggle Theme(Temp)
      </div>
      <div
        class="c5 select-none"
      >
        Toggle Mode(Temp)
      </div>
      <div
        class="c5"
      >
        <svg
          class="h-6 w-6"
          viewBox="0 0 513 342"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h513v342H0z"
            fill="#FFF"
          />
          <path
            d="M0 0h513v26.3H0zm0 52.6h513v26.3H0zm0 52.6h513v26.3H0zm0 52.6h513v26.3H0zm0 52.7h513v26.3H0zm0 52.6h513v26.3H0zm0 52.6h513V342H0z"
            fill="#D80027"
          />
          <path
            d="M0 0h256.5v184.1H0z"
            fill="#2E52B2"
          />
          <path
            d="m47.8 138.9-4-12.8-4.4 12.8H26.2l10.7 7.7-4 12.8 10.9-7.9 10.6 7.9-4.1-12.8 10.9-7.7zm56.3 0-4.1-12.8-4.2 12.8H82.6l10.7 7.7-4 12.8 10.7-7.9 10.8 7.9-4-12.8 10.7-7.7zm56.5 0-4.3-12.8-4 12.8h-13.5l11 7.7-4.2 12.8 10.7-7.9 11 7.9-4.2-12.8 10.7-7.7zm56.2 0-4-12.8-4.2 12.8h-13.3l10.8 7.7-4 12.8 10.7-7.9 10.8 7.9-4.3-12.8 11-7.7zM100 75.3l-4.2 12.8H82.6L93.3 96l-4 12.6 10.7-7.8 10.8 7.8-4-12.6 10.7-7.9h-13.4zm-56.2 0-4.4 12.8H26.2L36.9 96l-4 12.6 10.9-7.8 10.6 7.8L50.3 96l10.9-7.9H47.8zm112.5 0-4 12.8h-13.5l11 7.9-4.2 12.6 10.7-7.8 11 7.8-4.2-12.6 10.7-7.9h-13.2zm56.5 0-4.2 12.8h-13.3l10.8 7.9-4 12.6 10.7-7.8 10.8 7.8-4.3-12.6 11-7.9h-13.5zm-169-50.6-4.4 12.6H26.2l10.7 7.9-4 12.7L43.8 50l10.6 7.9-4.1-12.7 10.9-7.9H47.8zm56.2 0-4.2 12.6H82.6l10.7 7.9-4 12.7L100 50l10.8 7.9-4-12.7 10.7-7.9h-13.4zm56.3 0-4 12.6h-13.5l11 7.9-4.2 12.7 10.7-7.9 11 7.9-4.2-12.7 10.7-7.9h-13.2zm56.5 0-4.2 12.6h-13.3l10.8 7.9-4 12.7 10.7-7.9 10.8 7.9-4.3-12.7 11-7.9h-13.5z"
            fill="#FFF"
          />
        </svg>
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <svg>
            bell
          </svg>
        </div>
      </div>
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <svg>
            gear
          </svg>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c8 c9"
        />
        <div
          class="c10"
        >
          <span
            class="c11"
          >
            Admin
          </span>
          <span
            class="c12"
          >
            Tanawat P
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
